src/core/text/TextMetrics.js

```diff
@@ -0,0 +1,191 @@
+import Text from './Text';
```

Comment:
Does this actually work though with the circular dependency? Normally, when you have a circular import like this one of them just ends up being an empty object and it doesn't work (wichever comes second). I think this either needs to be removed or have unit tests proving it works.